src/lib/utils/ct_utils.h

```diff
@@ -250,28 +250,25 @@ template <unpoisonable T>
 */
 template <typename T>
 constexpr inline T value_barrier(T x) {
```

Comment:
Unrelated, but we might want to restrict the input types for this. Likely `std::integral`, or perhaps any trivial type? I'm not sure about the restrictions of the asm context, tbh.

Reply:
TBH I'm not sure if that would work or not. It is however probably not sensible to ever do so, even if it worked - if you need to `value_barrier` some complex `T` probably instead `T` should have had some constant time logic which internally used `value_barrier` on the integer types which `T` contains.

Reply:
I added a restriction that `CT::value_barrier` must be a non-boolean unsigned integral type.